From 739d1ab1c9907829e47990779215a8741e55bc31 Mon Sep 17 00:00:00 2001
From: Caesar Wang <wxt@rock-chips.com>
Date: Sat, 24 Feb 2018 11:29:24 +0800
Subject: [PATCH] arm64: dts: rockchip: enable the dmc for rk3399 sapphire
 excavator

The rk3399 had supported the dmc, the sapphire excavator board should
supported it too.

Change-Id: Id462ca1957b8c4960564d0ca24f71e7811aaabfd
Signed-off-by: Caesar Wang <wxt@rock-chips.com>
---
 .../rockchip/rk3399-excavator-sapphire.dtsi   | 31 ++++++++++++++++++
 .../rk3399-sapphire-excavator-edp.dts         | 32 -------------------
 2 files changed, 31 insertions(+), 32 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-excavator-sapphire.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-excavator-sapphire.dtsi
index 344b2a53c777..c406e54b4a87 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-excavator-sapphire.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-excavator-sapphire.dtsi
@@ -126,6 +126,37 @@
 	};
 };
 
+&dfi {
+	status = "okay";
+};
+
+&dmc {
+	status = "okay";
+	center-supply = <&vdd_center>;
+	upthreshold = <40>;
+	downdifferential = <20>;
+	system-status-freq = <
+		/*system status         freq(KHz)*/
+		SYS_STATUS_NORMAL       800000
+		SYS_STATUS_REBOOT       528000
+		SYS_STATUS_SUSPEND      200000
+		SYS_STATUS_VIDEO_1080P  200000
+		SYS_STATUS_VIDEO_4K     600000
+		SYS_STATUS_VIDEO_4K_10B 800000
+		SYS_STATUS_PERFORMANCE  800000
+		SYS_STATUS_BOOST        400000
+		SYS_STATUS_DUALVIEW     600000
+		SYS_STATUS_ISP          600000
+	>;
+	vop-bw-dmc-freq = <
+	/* min_bw(MB/s) max_bw(MB/s) freq(KHz) */
+		0       577      200000
+		578     1701     300000
+		1702    99999    400000
+	>;
+	auto-min-freq = <200000>;
+};
+
 &spdif {
 	status = "okay";
 	pinctrl-0 = <&spdif_bus>;
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dts b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dts
index ae154890b108..728b430079c0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-sapphire-excavator-edp.dts
@@ -118,38 +118,6 @@
 	enable-gpios = <&gpio4 29 GPIO_ACTIVE_HIGH>;
 };
 
-&dfi {
-	status = "disabled";
-};
-
-&dmc {
-	status = "disabled";
-	center-supply = <&vdd_center>;
-	upthreshold = <45>;
-	downdifferential = <20>;
-	system-status-freq = <
-		/*system status         freq(KHz)*/
-		SYS_STATUS_NORMAL       800000
-		SYS_STATUS_REBOOT       528000
-		SYS_STATUS_SUSPEND      200000
-		SYS_STATUS_VIDEO_1080P  200000
-		SYS_STATUS_VIDEO_4K     600000
-		SYS_STATUS_VIDEO_4K_10B 800000
-		SYS_STATUS_PERFORMANCE  800000
-		SYS_STATUS_BOOST        400000
-		SYS_STATUS_DUALVIEW     600000
-		SYS_STATUS_ISP          600000
-	>;
-	vop-bw-dmc-freq = <
-	/* min_bw(MB/s) max_bw(MB/s) freq(KHz) */
-		0       762      200000
-		763     1893     400000
-		1894    3012     528000
-		3013    99999    800000
-	>;
-	auto-min-freq = <200000>;
-};
-
 &edp {
 	status = "okay";
 	force-hpd;
-- 
2.35.3

